keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
+The intention of the proposed changes is to provide a supportable replacement for the current alpha `kubectl apply --prune` semantics.  Our intention is not to change the behavior of the existing `--prune` functionality, but rather to produce an alternative that users will happily and safely move to.  The two will be documented as separate pruning "modes". The proposal may evolve at the coding/PR stage, but the current plan is as follows.
 
-Required for an MVP release:
+The flag and any subcommands will be treated as alpha initially.  During alpha, users will need to set an environment variable, `KUBECTL_APPLYSET=1`, to make them available.
+
+Included in kubectl v1.27:
 - `KUBECTL_APPLYSET=1` environment variable: Required to expose the new flags/commands during alpha.
-- `kubectl apply --prune --applyset=[resource.version.group/]name`: The `--applyset` flag MUST be used with `--prune` and MUST have a non-empty value when used. Its GVR component is defaulted to `secrets` when missing. This flag CANNOT be used with `-l/--selector` or with `--prune-allow-list`, and this will be validated at flag parsing time.
-- `kubectl apply --prune --applyset=<id> --dry-run`
-- `kubectl diff --prune --applyset=<id>`
+- `kubectl apply --prune --applyset=[RESOURCE][.GROUP]/NAME`: The `--applyset` flag MUST be used with `--prune` and MUST have a non-empty value when used. Its GR component is defaulted to `secrets` when missing. This flag CANNOT be used with `-l/--selector` or with `--prune-allow-list`, and is validated at flag parsing time.
+- `kubectl apply --prune --applyset=[RESOURCE][.GROUP]/NAME --dry-run`
 
-Tentatively proposed for future iterations (more specific design details to follow after MVP):
-- `kubectl apply generate-applyset <id> --selector=[key=val] --legacy-allow-list=[]`: command to migrate from the legacy pruning system to this new one.
-- `kubectl apply verify-applyset [<id>] [--fix]`: `fsck`-style functionality to update the annotations on the parent ApplySet objects.
-- `kubectl apply view-applyset <id> -o name|json|yaml`: A command for viewing ApplySet membership, ideally in a way that can be programmatically chained.
-- `kubectl apply disband-applyset <id>`: removes the `applyset.k8s.io/id` from all members and then deletes the parent ApplySet object.
-- `kubectl apply list-applysets`: view apply sets, including those managed by other tools.
+To be included in v1.28:
+- `kubectl diff --prune --applyset=[RESOURCE][.GROUP]/NAME`
 
-Examples:
 
 ```bash
 # Simple namespace-scoped apply with prune.
 # The parent object will be a Secret named "set1" in the "foo" namespace.
 kubectl apply -n foo --prune --applyset=set1  -f .
 
-# Simple apply with prune, with cluster-scoped ApplySet
-# The parent object will be the "myapp" Namespace itself.
-kubectl apply -n myapp --prune --applyset=namespaces/myapp -f .
```

Comment:
we decided not to allow anything other than Secret/ConfigMap even during the last KEP update, and this seemingly got missed